### ISSUE TITLE
Error on mqtt-sdcard DEV

### ISSUE THREAD

**Issue by Deca** [xdk2mam/xdk2mam#8]
Hi again,

I'm trying to setup the mqtt-sdcard in the DEV repository
I've installed workbench 3.5, cleaned, compiled and flashed the mqtt-sdcard. Config.cfg parameters are set and double checked

After flashing I get this error:
INFO | XDK DEVICE 1: System Error:, Severity:2 Error code: 67 Package Id:10 and module ID is :2 

Do you have any suggestion?

**Comment by aleelus** [comment]
Hi @Deca,
Did you try to turn on the xdk with the micro sd (which contains the file 'config.cfg') inside?

**Comment by Deca** [comment]
Yes, I'm running the XDK with a 2gb sdcard and the config.cfg parameters edited for my network

On the xdk forum there is another user that have the same error:
https://xdk.bosch-connectivity.com/community/-/message_boards/message/352830

They suggest him that the package id:10 is related to the BSP package. I did a search for "PACKAGE_ID = 10" in the SDK folder but I can't find anything

**Comment by aleelus** [comment]
Hi @Deca,
In the configuration file there should be no space or tabs (unless it is part of a password, name of ssid, etc).

PS: Just in case, the name of the file must be 'config.cfg' without the capital letter C.

**Comment by Deca** [comment]
Yes I confirm that there aren't space or tabs of any kind and the file name is lowercase, here a screenshot of the .cfg and the error
![image](https://user-images.githubusercontent.com/3716783/54528892-95d54600-497e-11e9-9a30-58fc96853389.png)

**Comment by aleelus** [comment]
@Deca, can you try to put a '\n' or hit enter after the 'YES' of the accoustic sensor (this version of parser seeks the '\n')?

**Comment by Deca** [comment]
Added the newline char after the "YES" unfortunately I get the same error. I tried also with another sdcard (sandisk 16gb) just in case


**UPDATE:**
@aleelus  Could be an issue with the "zero" character and the end of a line? 

I wanted to give a try to another package with externall .cfg file, so I flashed the http-sdcard and I get the same error but this time the device console is more verbose:

`INFO | XDK DEVICE 1: WLANPersonalWPA2Connect : Connecting to ZyXEL_FD5 
 INFO | XDK DEVICE 1:  WLANPersonalWPA2Connect : Error in WlanConnect_WPA 
 INFO | XDK DEVICE 1: Error in WLAN package.
 INFO | XDK DEVICE 1: 	Package ID: 10
 INFO | XDK DEVICE 1: 	Module ID: 2
 INFO | XDK DEVICE 1: 	Severity code: 2
 INFO | XDK DEVICE 1: 	Error code: 67
 INFO | XDK DEVICE 1: WLAN_Enable failed with 167911491 `

It refer to my wifi SSID name without the 'zero' at the end "ZyXEL_FD5" , since the complete name is 'ZyXEL_FD50'

**UPDATE 2:**

I've tried with another wifi (mobile hotspot) and get the same connection error. The device console strip the last SSID character, so I'm not sure if is related to the connection error or just the printf output

**Comment by Deca** [comment]
I finally managed to get a working configuration. 
I had to open the .cfg file with an editor able to show hidden characters and put a carriage return (0x0D) before the line feed for each line
This is beyond to my understanding since I barely have a basic C language background, I just know that working with buffer and binary mode could be really tricky... @aleelus if you don't mind could you explain what is going on exactly?

![image](https://user-images.githubusercontent.com/3716783/54548085-671f9580-49a7-11e9-95f2-3a18ea4ff1ae.png)

**Comment by aleelus** [comment]
@Deca,
Thanks for notice this issue, in fact text files created on DOS/Windows machines have different line endings than files created on Unix/Linux. DOS uses carriage return and line feed ("\r\n") as a line ending, which Unix uses just line feed ("\n").
You can change the format in Notepad++ if you go to Edit->EOL Conversion->Windows (CR LF)
